app/src/main/java/lt/vilnius/tvarkau/fragments/NewReportFragment.kt

```diff
@@ -0,0 +1,573 @@
+        } else {
+            activity.onBackPressed()
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        EasyImage.handleActivityResult(requestCode, resultCode, data, activity, object : DefaultCallback() {
+            override fun onImagePickerError(e: Exception?, source: EasyImage.ImageSource?, type: Int) {
+                Toast.makeText(activity, R.string.photo_capture_error, Toast.LENGTH_SHORT).show()
+                Timber.w(e, "Unable to take a picture")
+            }
+
+            override fun onImagesPicked(imageFiles: List<File>, source: EasyImage.ImageSource, type: Int) {
+                setImagesInViewPager(imageFiles)
+            }
+
+            override fun onCanceled(source: EasyImage.ImageSource?, type: Int) {
+                if (source == EasyImage.ImageSource.CAMERA) {
+                    val photoFile = EasyImage.lastlyTakenButCanceledPhoto(context)
+                    photoFile?.delete()
+                }
+            }
+        })
+
+        if (resultCode == Activity.RESULT_OK) {
+            when (requestCode) {
+                REQUEST_PLACE_PICKER -> {
+                    val geocoder = Geocoder(context)
+                    val place = PlacePicker.getPlace(context, data)
+
+                    var addresses = listOf<Address>()
+
+                    place.latLng?.let {
+                        locationCords = it
+
+                        try {
+                            addresses = geocoder.getFromLocation(it.latitude, it.longitude, 1)
+                        } catch (e: IOException) {
+                            Timber.e(e)
+                        }
+                    }
+
+                    if (addresses.isNotEmpty()) {
+                        if (addresses.first().locality != null) {
+                            val address = addresses[0].getAddressLine(0)
+                            report_problem_location_wrapper.error = null
+                            report_problem_location.setText(address)
+                        }
+                    } else {
+                        // Mostly when Geocoder throws IOException
+                        // backup solution which in not 100% reliable
+                        val addressSlice = place.address.toString()
+                                .split(", ".toRegex())
+                                .dropLastWhile(String::isEmpty)
+                                .toTypedArray()
+
+                        val address = addressSlice[0]
+                        report_problem_location_wrapper.error = null
+                        report_problem_location.setText(address)
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle?) {
+        super.onSaveInstanceState(outState)
+        outState?.let {
+            it.putParcelable(SAVE_LOCATION, locationCords)
+            it.putSerializable(SAVE_PHOTOS, imageFiles)
```

Comment:
Yup there is issue https://github.com/frankiesardo/icepick/issues/47 with workaround to make KT work with Icepick. IMHO it's better to not use icepick with KT